pkg/providers/aws/provider_postgres.go

```diff
@@ -0,0 +1,487 @@
+			return v1alpha1.StatusMessage(msg), errorUtil.Wrapf(err, msg)
+		}
+		return "", nil
+	}
+
+	// check if instance db exists and is available
+	if *foundInstance.DBInstanceStatus == "available" {
+		if *foundInstance.DeletionProtection {
+			_, err = instanceSvc.ModifyDBInstance(&rds.ModifyDBInstanceInput{
+				DBInstanceIdentifier: postgresDeleteConfig.DBInstanceIdentifier,
+				DeletionProtection:   aws.Bool(false),
+			})
+			if err != nil {
+				msg := "waiting on AWS ModifyDBInstance permission"
+				return v1alpha1.StatusMessage(msg), errorUtil.Wrap(err, msg)
+			}
+			return "turning off deletion protection", nil
+		}
+		_, err = instanceSvc.DeleteDBInstance(postgresDeleteConfig)
+		postgresErr, isAwsErr := err.(awserr.Error)
+		if err != nil && !isAwsErr {
+			msg := fmt.Sprintf("failed to delete rds instance : %s", err)
+			return v1alpha1.StatusMessage(msg), errorUtil.Wrapf(err, msg)
+		}
+		if err != nil && isAwsErr {
+			if postgresErr.Code() != rds.ErrCodeDBInstanceNotFoundFault {
+				msg := fmt.Sprintf("failed to delete rds instance : %s", err)
+				return v1alpha1.StatusMessage(msg), errorUtil.Wrapf(err, msg)
+			}
+		}
+	}
+	return "postgres instance deletion in progress", nil
+}
+
+// function to get rds instances, used to check/wait on AWS credentials
+func getPostgresInstances(cacheSvc rdsiface.RDSAPI) ([]*rds.DBInstance, error) {
+	var pi []*rds.DBInstance
+	err := wait.PollImmediate(time.Second*5, time.Minute*5, func() (done bool, err error) {
+		listOutput, err := cacheSvc.DescribeDBInstances(&rds.DescribeDBInstancesInput{})
+		if err != nil {
+			return false, nil
+		}
+		pi = listOutput.DBInstances
+		return true, nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return pi, nil
+}
+
+func (p *AWSPostgresProvider) getPostgresConfig(ctx context.Context, r *v1alpha1.Postgres) (*rds.CreateDBInstanceInput, *rds.DeleteDBInstanceInput, *StrategyConfig, error) {
+	stratCfg, err := p.ConfigManager.ReadStorageStrategy(ctx, providers.PostgresResourceType, r.Spec.Tier)
+	if err != nil {
+		return nil, nil, nil, errorUtil.Wrap(err, "failed to read aws strategy config")
+	}
+	if stratCfg.Region == "" {
+		stratCfg.Region = DefaultRegion
+	}
+
+	postgresCreateConfig := &rds.CreateDBInstanceInput{}
+	if err := json.Unmarshal(stratCfg.RawStrategy, postgresCreateConfig); err != nil {
+		return nil, nil, nil, errorUtil.Wrap(err, "failed to unmarshal aws postgres cluster configuration")
+	}
+
+	postgresDeleteConfig := &rds.DeleteDBInstanceInput{}
+	if err := json.Unmarshal(stratCfg.DeleteStrategy, postgresDeleteConfig); err != nil {
+		return nil, nil, nil, errorUtil.Wrap(err, "failed to unmarshal aws postgres cluster configuration")
+	}
+	return postgresCreateConfig, postgresDeleteConfig, stratCfg, nil
+}
+
+// verifies if there is a change between a found instance and the configuration from the instance strat
+func verifyPostgresChange(postgresConfig *rds.CreateDBInstanceInput, foundConfig *rds.DBInstance) *rds.ModifyDBInstanceInput {
```

Comment:
Maybe `buildPostgresUpdateStrategy`? This would follow the same convention as `buildPostgresCreateStrategy` (and `buildPostgresDeleteStrategy` and infer that we're changing stuff rather than checking stuff